src/pages/Admin/Charity/Invest/Investor/LockDuration.tsx

```diff
@@ -0,0 +1,54 @@
+import { UseFormRegisterReturn, useFormContext } from "react-hook-form";
+import { FormValues } from "./types";
+
+export default function LockDuration({ classes = "" }) {
+  const { register, watch } = useFormContext<FormValues>();
+  const type = watch("type");
+
+  if (type === "liquid") return null;
+
+  return (
+    <div className={`grid sm:grid-cols-3 gap-3 ${classes}`}>
+      <h3 className="col-span-full font-bold">Choose lock duration</h3>
+      <LockOption register={register("lockPeriod")} id="duration" value={"1"} />
+      <LockOption register={register("lockPeriod")} id="duration" value={"7"} />
+      <LockOption
+        register={register("lockPeriod")}
+        id="duration"
+        value={"14"}
+      />
+    </div>
+  );
+}
+
+type OptionProps = {
+  register: UseFormRegisterReturn;
+  value: FormValues["lockPeriod"];
+  id: string;
+};
+
+function LockOption({ register, id, value }: OptionProps) {
+  return (
+    <div className="grid grid-cols-[auto_1fr] gap-x-4 gap-y-0.5 items-center py-3 px-[1.1125rem] rounded border border-prim">
+      <input
+        {...register}
+        id={id + value}
```

Comment:
nit: instead of `id={id + value}`, the `id` value could immediately be set to the desired value in the parent component.